packages/ui-components/src/components/PopupMenu/PopupMenu.component.tsx

```diff
@@ -0,0 +1,394 @@
+              </Float.Reference>
+
+              {/* Render the menu in our portal: */}
+              <PortalProvider.Portal>
+                <Float.Content>
+                  {/* Wrap the floated content in a div that headless ui Float.Content can reference */}
+                  <div className={`juno-popupmenu-float-content-wrapper`}>{menu}</div>
+                </Float.Content>
+              </PortalProvider.Portal>
+            </Float>
+          </PopupMenuContext.Provider>
+        )
+      }}
+    </HeadlessMenu>
+  )
+}
+
+// TOGGLE COMPONENT
+const PopupMenuToggle: React.FC<PopupMenuToggleProps> = ({
+  as = "button",
+  disabled = false,
+  children = null,
+  className = "",
+  ...props
+}) => (
+  <HeadlessMenu.Button
+    as={as}
+    className={`juno-popupmenu-toggle ${disabled && disabledToggleStyles} ${className}`}
+    disabled={disabled}
+    {...props}
+  >
+    {children}
+  </HeadlessMenu.Button>
+)
+
+// MENU COMPONENT
+const PopupMenuMenu: React.FC<PopupMenuMenuProps> = ({ children = null, className = "", ...props }) => {
+  // Consume context to get the size to render and the close function:
+  const { menuSize } = usePopupMenuContext()
+  return (
+    <HeadlessMenu.Items
+      className={`juno-popupmenu-menu juno-popupmenu-menu-size-${menuSize} ${menuStyles} ${className}`}
+      {...props}
+    >
+      {children}
+    </HeadlessMenu.Items>
+  )
+}
+
+// ITEM COMPONENT
+const PopupMenuItem: React.FC<PopupMenuItemProps> = ({
+  as = "div",
+  children = null,
+  className = "",
+  disabled = false,
+  href,
+  icon = null,
+  label = "",
+  rel,
+  target,
+  ...props
+}) => {
+  // Consume context to get the size to render:
+  const { menuSize } = usePopupMenuContext()
+  // Determine the appropriate set of styles per size:
+  const itemSizeStyles = menuSize === "small" ? smallItemStyles : normalItemStyles
+  // Determine the compontn to render as:
+  return (
+    <HeadlessMenu.Item
+      as={as}
+      disabled={disabled}
+      className={`juno-popupmenu-item ${itemStyles} ${disabled ? disabledItemStyles : actionableItemStyles} ${itemSizeStyles} ${className}`}
+      {...(as === "a" ? { href, rel, target } : {})} // Conditionally spread anchor-specific props only when renering an anchor
+      {...props}
+    >
+      {/* 
+      To render the item content, provide an itemBag context to the headless ui API. This includes properties like active, disabled, close. 
+      Then, if label is a string, render it. Otherwise, if children is a function, invoke it with itemBag; otherwise render children straightaway:
+    */}
+      {(itemBag) => (
+        <>
+          {icon && <Icon icon={icon} size="18" className={`juno-popupmenu-item-icon ${itemIconStyles}`} />}
+          {/* Make sure to test for truthy label and then check for string type, otherwise typeof label === string would alway return true due to the default empty string still being of type string: */}
+          {label && typeof label === "string"
```

Comment:
(Question)
I would not follow the guides in this case that we used to follow everywhere and try to think maybe in a advance here, since the component is new. 
Why should we support string ? why should we send children as a function? or other way around, is it the way to avoid this logic and checks? Open for discussion

Reply:
To stay consistent with all our other components, we need to accept both string and children. A user may pass a function as children in order to make use of the render props (`itemBag`) Headless Menu provides per child (see comment in the code). Not handling this case will fail and produce errors.